pkg/domain/stateDiffComponent.go

```diff
@@ -48,6 +48,19 @@ type ComponentDiffState struct {
 	DeployConfig ecosystem.DeployConfig
 }
 
+// IsExpectedVersion checks if the given version es equal to the expected version
+func (diff ComponentDiff) IsExpectedVersion(actualVersion *semver.Version) bool {
```

Comment:
Wouldn't `NeedsUpgrade` be a better name? Of course, the booleans would need to be reversed, then.

Reply:
This function is just a helper function to compare the expectedVersion in the diff with the given version. It is not exclusively used to determine if we need to update (for this `HasChanges()` is used).

We use this function twice:
1. to check if the `spec.version` of a component is already correct (triggers upgrade)
2. to check if the actual installed version in the component already matches the wanted version (acts like a health check, no further upgrade will happen here)